src/main/kotlin/org/rust/lang/core/types/infer/TypeInferenceWalker.kt

```diff
@@ -654,11 +654,21 @@ class RsTypeInferenceWalker(
 
         val constParameters = callee.element.constParameters.map { CtConstParameter(it) }
         val resolver = PathExprResolver.fromContext(ctx)
-        val constArguments = methodCall.constArguments.withIndex().map { (i, expr) ->
-            val expectedTy = constParameters.getOrNull(i)?.parameter?.typeReference?.type ?: TyUnknown
-            expr.evaluate(expectedTy, resolver)
-        }
-        val constSubst = constParameters.zip(constArguments).toMap()
+        val constSubst = constParameters.zip(methodCall.constArguments).mapNotNull { (param, expr) ->
+            val expectedTy = param.parameter.typeReference?.type ?: TyUnknown
+            param to when (expr) {
```

Comment:
This looks like a duplication of code in `RsPsiSubstitution.toSubst`. Is it not possible to deduplicate it somehow?